torch/csrc/jit/passes/quantization.cpp

```diff
@@ -503,8 +503,56 @@ void InsertQuantDequantNodesForParam(
   }
 }
 
+void InsertQuantDequantNodesForParam(
+    script::Method& method,
+    const std::string& param_name,
+    const std::function<std::tuple<std::string, float, int>(float, float)>&
+        getQParamFunc,
+    at::ScalarType t) {
+  AT_ASSERT(getQParamFunc != nullptr);
+  auto params_to_insert_qdq = getQuantizableParamsofName(method, param_name);
+
+  for (auto& param_info : params_to_insert_qdq) {
```

Comment:
I know we've had a discussion on using of `auto`, but here it really obfuscates the code - I have no idea what the type of `param_info` is.

Also, do we really expect to ever see more than one parameter for the given name? Maybe assert that there is at most one parameter with the given name inside `getQuantizableParamsofName ` and return it instead of a vector?

Reply:
So eventually what will happen is, as we expand to different quant schemes, we will have more cases here for weight and bias so more function signature is unavoidable. The interface will continue to stay same because it takes in template as function.

Else other approach was to have explicit pass for weight and bias but not using template would lead to explosion of torch apis in quantization.h

Reply:
Lets discuss once u  r back in detail

Reply:
Answer to first comment, yeah its possible to have multiple weights as inputs to multiple conv to the top level module. I believe this part is going to change though during lock-down how params are represented in lowered graphs